internal/bminventory/inventory.go

```diff
@@ -6218,9 +6223,18 @@ func (b *bareMetalInventory) V2DownloadClusterCredentials(ctx context.Context, p
 	// At the finalizing phase, we create the kubeconfig file and add the ingress CA.
 	// An ingress CA isn't required for normal login but for oauth login which isn't a common use case.
 	// Here we fallback to the kubeconfig-noingress for the kubeconfig filename.
-	if err != nil && params.FileName == constants.Kubeconfig {
+	if err != nil && fileName == constants.Kubeconfig {
 		fileName = constants.KubeconfigNoIngress
-		respBody, contentLength, err = b.v2DownloadClusterFilesInternal(ctx, constants.KubeconfigNoIngress, params.ClusterID.String())
+
+		if b.Config.ClusterTLSCertOverrideDir != "" {
```

Comment:
Is there anything preventing certs re-generation if called twice? I was expecting also a more strict check condition, ie Invoker == agent && ClusterTLSCertOverrideDir empty